build/docker/Dockerfile.centos6

```diff
@@ -13,33 +13,47 @@ RUN yum install -y centos-release-scl yum-utils && yum-config-manager --enable r
 # Extra repos & dependencies
 RUN yum update -y && yum install -y epel-release
 
-RUN yum install -y which wget sudo make tree vim cmake zip git redhat-lsb \
+RUN yum install -y which wget sudo make tree vim cmake zip redhat-lsb \
     openssh-clients bind-utils bison gcc-c++ libcxx libstdc++-static rpm-devel \
     pam-devel openssl-devel rpm-build mysql-devel curl-devel selinux-policy-devel \
     audit-libs-devel boost148-devel lsof net-tools patch curl tar bzip2 unzip \
     && yum clean all
- 
+
 # Devtoolset-2 (gcc 4.8)
-RUN wget http://people.centos.org/tru/devtools-2/devtools-2.repo -O /etc/yum.repos.d/devtools-2.repo 
+RUN wget http://people.centos.org/tru/devtools-2/devtools-2.repo -O /etc/yum.repos.d/devtools-2.repo
 RUN yum update -y && yum install -y devtoolset-2-gcc devtoolset-2-gcc-c++ devtoolset-2-binutils
 ENV PATH /opt/rh/devtoolset-2/root/usr/bin:$PATH
 
-# Ruby 2.4
-ADD https://cache.ruby-lang.org/pub/ruby/2.4/ruby-2.4.4.tar.gz /home/scratch/ruby-2.4.4.tar.gz
-RUN cd /home/scratch && tar -zxf ruby-2.4.4.tar.gz
-RUN cd /home/scratch/ruby-2.4.4 && ./configure && make && make install
+# git
+ADD https://github.com/git/git/archive/v2.17.1.tar.gz /home/scratch/v2.17.1.tar.gz
+RUN tar -xzf v2.17.1.tar.gz
+RUN cd git-2.17.1/ && make configure && ./configure --prefix=/usr && make all && make install
 
 # OpenSSL
 RUN mkdir -p /home/scratch/ostc-openssl
 RUN mkdir -p ~/.ssh/ && ssh-keyscan github.com >> ~/.ssh/known_hosts
-RUN git clone --branch OpenSSL_1_0_0-stable  https://github.com/openssl/openssl.git /home/scratch/ostc-openssl/openssl-1.0.0
-RUN git clone --branch OpenSSL_1_1_0-stable  https://github.com/openssl/openssl.git /home/scratch/ostc-openssl/openssl-1.1.0
+RUN git clone --branch OpenSSL_1_0_0 https://github.com/openssl/openssl.git /home/scratch/ostc-openssl/openssl-1.0.0
```

Comment:
why not stable

Reply:
stable branch will point to releases with 1.0.0a...1.0.0e which is not backward compatible with 1.0.0, we need to target to older commit for that 1.0.0, 1.0.1 and 1.1.0 branches.